src/org/zaproxy/zap/extension/viewstate/ViewStateModel.java

```diff
@@ -0,0 +1,328 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.viewstate;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.extension.encoder.Base64;
+import org.parosproxy.paros.network.HtmlParameter;
+import org.parosproxy.paros.network.HtmlParameter.Type;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.httppanel.view.hex.HttpPanelHexModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.AbstractHttpByteHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.HttpPanelViewModelUtils;
+import org.zaproxy.zap.extension.viewstate.zap.utils.ASPViewState;
+import org.zaproxy.zap.extension.viewstate.zap.utils.JSFViewState;
+import org.zaproxy.zap.extension.viewstate.zap.utils.ViewState;
+import org.zaproxy.zap.model.StandardParameterParser;
+
+import net.htmlparser.jericho.Attribute;
+import net.htmlparser.jericho.Attributes;
+import net.htmlparser.jericho.Element;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+
+public class ViewStateModel extends AbstractHttpByteHttpPanelViewModel {
+
+	private static Logger logger = Logger.getLogger(ViewStateModel.class);
+	private ArrayList<ViewState> viewstateParams = new ArrayList<ViewState>();
+	private ViewStateUpdatedListener vsListener;
+	private String viewstateType;
+	private int modelAction;
+	
+	public ViewStateModel(int action, ViewStateUpdatedListener listener) {
+		// Setup ViewState params to look for
+        viewstateParams.add(new ViewState(null, "ASP", "__VIEWSTATE"));
+        // For chunked ASP ViewStates
+        viewstateParams.add(new ViewState(null, "ASP", "__VIEWSTATEFIELDCOUNT"));
+        // For JSF ViewStates
+        viewstateParams.add(new ViewState(null, "JSF", "javax.faces.ViewState")); 
+        // Set the action for the model (request/response)
+        modelAction = action;
+        // Set the ViewState listener
+        vsListener = listener;
+	}
+	
+	@SuppressWarnings("unchecked")
+	public <T> T getViewStateParam(String body) {
+
+		T viewState = null;
+		for (ViewState vsp : viewstateParams) {
+			String val = getParamValue(body, vsp.getName());
+			if (val != null) {
+				logger.debug("Found ViewState param: " + vsp.getName() + ". Type: " + vsp.getType());
+				if (modelAction == ExtensionHttpPanelViewStateView.VS_ACTION_REQUEST) {
```

Comment:
The model should define these constants instead, to not have the model depend on the extension.